lib/Modules/ComputeHooks/ComputeHooks.psm1

```diff
@@ -106,1360 +56,800 @@ function Install-Prerequisites {
+        $ctxt['log_dir'] = "$NOVA_DEFAULT_LOG_DIR"
     }
-    $config = gc $service["template"]
-    $incompleteContexts = [System.Collections.Generic.List[object]](New-Object "System.Collections.Generic.List[object]")
-    $allContexts = [System.Collections.Generic.List[object]](New-Object "System.Collections.Generic.List[object]")
-    $mergedContext = [System.Collections.Generic.Dictionary[string, object]](New-Object "System.Collections.Generic.Dictionary[string, object]")
-
-    foreach ($context in $service['context_generators']){
-        Write-JujuInfo ("Getting context for {0}" -f $context["relation"])
-        $allContexts.Add($context["relation"])
-        $ctx = & $context["generator"]
-        Write-JujuInfo ("Got {0} context: {1}" -f @($context["relation"], $ctx.Keys))
-        if (!$ctx.Count){
-            # Context is empty. Probably peer not ready
-            Write-JujuWarning ("Context for {0} is EMPTY" -f $context["relation"])
-            $incompleteContexts.Add($context["relation"])
-            $should_restart = $false
-            continue
-        }
-        foreach ($val in $ctx.Keys) {
-            $mergedContext[$val] = $ctx[$val]
-        }
+    if(!$ctxt['instances_dir']) {
+        $ctxt['instances_dir'] = "$NOVA_DEFAULT_INSTANCES_DIR"
     }
-    Set-IncompleteStatusContext -ContextSet $allContexts -Incomplete $incompleteContexts
-    if(!$mergedContext.Count) {
-        return $false
+
+    foreach($dir in @($ctxt['log_dir'], $ctxt['instances_dir'])) {
+        if (!(Test-Path $dir)) {
+            New-Item -ItemType Directory -Path $dir
+        }
     }
-    $cfg = Invoke-RenderTemplateFromFile -Context $mergedContext -Template $service["template"]
-    Set-Content $service["config"] $cfg
-    return $should_restart
-}
 
-function Get-FallbackNetadapter {
-    $name = Get-MainNetadapter
-    $net = Get-NetAdapter -Name $name
-    return $net
+    return $ctxt
 }
 
-function Get-InterfaceFromConfig {
-    Param(
-        [string]$ConfigOption="data-port",
-        [switch]$MustFindAdapter=$false
-    )
-
-    $nic = $null
-    $DataInterfaceFromConfig = Get-JujuCharmConfig -Scope $ConfigOption
-    Write-JujuInfo "Looking for $DataInterfaceFromConfig"
-    if (!$DataInterfaceFromConfig){
-        if($MustFindAdapter) {
-            Throw "No data-port was specified"
-        }
-        return $null
-    }
-    $byMac = @()
-    $byName = @()
-    $macregex = "^([a-f-A-F0-9]{2}:){5}([a-fA-F0-9]{2})$"
-    foreach ($i in $DataInterfaceFromConfig.Split()){
-        if ($i -match $macregex){
-            $byMac += $i.Replace(":", "-")
-        }else{
-            $byName += $i
-        }
+function Uninstall-Nova {
+    $params = @{
+        'Namespace' = "novahyperv"
+        'Key' = "product_name_installed"
     }
-    if ($byMac.Length){
-        $nicByMac = Get-NetAdapter | Where-Object { $_.MacAddress -in $byMac -and $_.DriverFileName -ne "vmswitch.sys" }
+    $productName = Get-CharmState @params
```

Comment:
I would prefer we get this info from WMI. Cross check with known product names. See above comments on this matter.